### PR TITLE
Do not strip the `dylink.0` section from `libdl.so`

### PR DESCRIPTION
**Pull request by kateinoigakukun**
The `dylink.0` section is required to correctly layout the main memory space and provide `__memory_base` and `__table_base` symbols. The section was being stripped by the `strip` tool, and it led to the unpredictable corruption of `LIBRARIES` global variable.

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
I've tried to write test cases to kill this case, but I couldn't make a maintainable one 😢

**Comment by kateinoigakukun** [comment]
CC: @dicej

File: crates/wit-component/dl/build.sh [hunk 1]
```diff
@@ -7,4 +7,4 @@
 
 CARGO_PROFILE_RELEASE_LTO=true RUSTFLAGS="-C relocation-model=pic" cargo build --release --target=wasm32-wasi
 $WASI_SDK_PATH/bin/clang -shared -o $1 -Wl,--whole-archive ../../../target/wasm32-wasi/release/libdl.a -Wl,--no-whole-archive
-cargo run --manifest-path ../../../Cargo.toml -- strip -a $1 -o $1
+cargo run --manifest-path ../../../Cargo.toml -- strip $1 -o $1
```
Comment (dicej) [inline crates/wit-component/dl/build.sh:10]: Optional idea: we could run `strip` a second time using `--delete name`, which would remove the `name` section but keep the others.

**Review by dicej — APPROVED** [review]
LGTM -- thanks!

File: crates/wit-component/dl/build.sh (hunk 1 above)
Comment (kateinoigakukun) [reply crates/wit-component/dl/build.sh:10]: Sounds reasonable to me 👍

**Comment by kateinoigakukun** [comment]
Thanks!

**Pull request merged**